### DISCUSSION TITLE
Proposal: Move migration guides to a wiki

### DISCUSSION THREAD

**Discussion by bryevdv** [bokeh/bokeh#12476] [Documentation]
@tcmetzger, @mattpap, @pavithraes, @ianthomas23, @philippjfr I would like to propose a change to how we publish migration guides. I would like to move all them to a wiki page, and then link a section header  (automatically) from the release notes. This is for a few reasons:

* The releases page is getting extremely long (too long). 
* Putting migration guides in a wiki page means we can easily add/update migration information anytime after a release, in case things come to light after the release that we missed. Republishing old docs is a huge, huge hassle. 

Edit: here is what a Wiki page might look like (partial)

https://github.com/bokeh/bokeh/wiki/Migration-Guides

I would also like to standardized the exact list of subs-sections that each section (version) can have. This will help users with navigating the guides. 

Note that is is possible to clone wikis:

https://docs.github.com/en/communities/documenting-your-project-with-wikis/adding-or-editing-wiki-pages#cloning-wikis-to-your-computer

So there is no limitation to the web UI and it is still possible to collaborate on branches, etc. 

I am willing to undertake the work to convert all the old migration guides as well as update the `bokeh-releases` sphinx extension to add a link to the relevant section automatically to the release notes on the docs.

**Comment by ianthomas23** [comment]
I am absolutely fine with this.

**Comment by bryevdv** [comment]
@mattpap do you have any comments? In another PR you said "let's continue on the wiki" but it was not clear it that as just for this release or if you were signaling support for this proposal.

**Reply by mattpap** [reply]
I'm generally in favour. However, I think the the main release/migration page should be just links to detailed pages. For now I will be updating https://github.com/bokeh/bokeh/wiki/Bokeh-3.0-Release-Notes---Migration-Guide-Draft. Also, I assume migration to wiki means that `docs/releases/*.rst` will be removed or just `3.0.0.rst`?

**Reply by mattpap** [reply]
Or actually I'm not sure what I should  be editing, because there two quite different release/migration guides right now.

**Reply by bryevdv** [reply]
The proposal is only to move the migration guide subsection to the wiki. The releases page will still collect high level narrative overview of each release (e.g. the short bullet list) as well as links to relevant resources for that release (versioned docs, migration guide wiki section, SRI hashes, etc)

**Comment by bryevdv** [comment]
@bokeh/core what does everyone think of limiting migration guides on the wiki to 1.0 and later? I think at this point we could decide to regard the  0.x releases to be of only historical interest, and if the information is actually desired, the old docs pages will still have it.

**Comment by bryevdv** [comment]
This work has been completed
